### ISSUE TITLE
ternary evaluates all values before returning the chosen value

### ISSUE THREAD

**Issue by sgandon** [Masterminds/sprig#271]
```
$list := list "foo" "bar"
$third := gt (len $list) 2 | ternary (index $list 2) ""

```
This code will create an `reflect: slice index out of range` exception as it is evaluating each arguments of the ternary.
This wold be great to ony evaluate the one that need to be returned.

**Comment by moorereason** [comment]
This behavior is inherent to the Go template engine, not sprig.  All arguments are evaluated and then passed to `ternary`.

**Comment by sgandon** [comment]
yes I thought so, isn't there a way to avoid that ?

**Comment by moorereason** [comment]
Yes, don't use `ternary` in that situation.  😄 You must protect the `index $list 2` with an if-statement or something.  You really can't control the call flow with `ternary`.

**Comment by mattfarina** [comment]
Unfortunately, this is a product of the Go template engine. It's not something we can change in sprig.
